src/Microsoft.AspNetCore.Routing/Template/TemplateSegment.cs

```diff
@@ -18,7 +18,12 @@ public TemplateSegment()
 
         public TemplateSegment(RoutePatternPathSegment other)
         {
-            Parts = new List<TemplatePart>(other.Parts.Select(s => new TemplatePart(s)));
+            var partCount = other.Parts.Count;
```

Comment:
This looks good but while you're here you could add:

```cs
if (other == null)
{
    throw new ArgumentNullException(nameof(other));
}
```

Reply:
Done